### PR TITLE
fix: gocyclo issues in go report

### LINKED ISSUE #964: [GoReport Issue] fix issue for Pouch reported from GoReport
### Ⅰ. Issue Description

As an open source project, Pouch is emphasizing much on project quality including code and test and documents. 

In code part, we use tool go report to check the quality, which is concluded on https://goreportcard.com/report/github.com/alibaba/pouch.

Here is the entrance which could be clicked:

image

There are still some work we need to do for Pouch, although Pouch has got a score of A+:
image

Here, we sincerely invite warm-hearted contributors to help solve this issues. It is really important to improve pouch on all these part. 🌹 🦁 🍺 🌞

### PR DESCRIPTION
**Pull request by zhuangqh**
### Ⅰ. Describe what this PR did
1. fix the **gocyclo** part of goreport 
2. fix bug: the no-proxy's environment name should be 'NO_PROXY' or 'no_proxy', not 'noProxy'.

### Ⅱ. Does this pull request fix one issue?

fixes #964

### CODE REVIEW

**Comment by allencloud** [comment]
Thanks a lot for your contribution. @zhuangqh 
Also cc @zeppp Could you help to review this?

File: ctrd/image_proxy_util.go [hunk 1]
```diff
@@ -66,7 +66,7 @@ func canonicalAddr(url *url.URL) string {
 // change no_proxy to noProxy for avoid fail in go lint check
 var (
 	noProxyEnv = &envOnce{
-		names: []string{"NO_PROXY", "noProxy"},
+		names: []string{"NO_PROXY", "no_proxy"},
```
Comment (allencloud) [inline ctrd/image_proxy_util.go]: Could you help to confirm this? Since I think it is introduced by your part. @Ace-Tang

**Comment by zeppp** [comment]
LGTM

File: ctrd/image_proxy_util.go (hunk 1 above)
Comment (allencloud) [reply ctrd/image_proxy_util.go]: have communicated with @Ace-Tang offline. The change is totally reasonable.

**Comment by allencloud** [comment]
LGTM

**Comment by allencloud** [comment]
Thank @zhuangqh for your first contribution.

**Comment by allencloud** [comment]
Oh, I have forgotten to squash the four commits into a single one. My fault.
And please sumbit a pull request with only a single commit in the following pull request. Thanks a lot.

**Comment by zhuangqh** [comment]
Sorry my fault. I have read the contributing guide carefully. 😅 @allencloud

**Pull request merged**